runtime/CMakeLists.txt

```diff
@@ -26,11 +26,15 @@ foreach(DF ${DOCFILES})
   list(APPEND BUILDDOCFILES ${GENERATED_RUNTIME_DIR}/doc/${F})
 endforeach()
 
-add_custom_command(OUTPUT ${BUILDDOCFILES}
-  COMMAND ${CMAKE_COMMAND} -E copy_directory
-    ${PROJECT_SOURCE_DIR}/runtime/doc ${GENERATED_RUNTIME_DIR}/doc
-  DEPENDS
-    ${DOCFILES})
+set(DOCFILE_DEPS ${DOCFILES})
+foreach(DOCFILE ${BUILDDOCFILES})
+  list(REMOVE_AT DOCFILE_DEPS 0)
+  add_custom_command(OUTPUT ${DOCFILE}
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+      ${PROJECT_SOURCE_DIR}/runtime/doc ${GENERATED_RUNTIME_DIR}/doc
+    DEPENDS
+    ${DOCFILE_DEPS})
+endforeach()
```

Comment:
Two things.  First, your repeatedly copying all the files into the directory over and over again--I don't think we want that.

Second, perhaps the better thing to do would be make each docfile depend on it's original source file, instead of this set that gradually dwindles?  We could pitch this loop, and re-use the one above:

```cmake
set(BUILDDOCFILES)
foreach(DF ${DOCFILES})
  get_filename_component(F ${DF} NAME)
  set(BF ${GENERATED_RUNTIME_DIR}/doc/${F})
  list(APPEND BUILDDOCFILES ${BF})
  add_custom_command(OUTPUT ${BF}
    COMMAND ${CMAKE_COMMAND} -E copy ${DF} ${BF}
    DEPENDS ${DF})
endforeach()
```

I think something like this would be better, IMHO, since we're only copying the source file and nothing more for a particular output.